sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -201,6 +201,23 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
+  test("SPARK-22226: group splitted expressions into one method per nested class") {
```

Comment:
Besides the unit test, can you provide an end-to-end case that can trigger this issue too?

Reply:
I have a use case where I faced this problem. And I tried this patch on it. Unfortunately this contains a very complex business logic and I have not been able to reproduce it in a simple one. But if needed, I can try again.

Reply:
Instead of copying your customer codes, can you making a fake one?

Reply:
@mgaido91 I can reproduce the issue by following test case. You can check it:

```scala
  test("SPARK-22226: too much splitted expressions should not exceen constant pool limit") {
    withSQLConf(
      (SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key, "false")) {
      val colNumber = 1000
      val baseDF = spark.range(10).toDF()
      val newCols = (1 to colNumber).map { colIndex =>
        expr(s"id + $colIndex").as(s"_$colIndex")
      }
      val input = baseDF.select(newCols: _*)
      val aggs2 = (1 to colNumber).flatMap { colIndex =>
        val colName = s"_$colIndex"
        Seq(expr(s"stddev($colName)"),
          expr(s"stddev_samp($colName)"),
          expr(s"stddev_pop($colName)"),
          expr(s"variance($colName)"),
          expr(s"var_samp($colName)"),
          expr(s"var_pop($colName)"),
          expr(s"skewness($colName)"),
          expr(s"kurtosis($colName)"))
      }
      input.agg(aggs2.head, aggs2.tail: _*).collect()
    }
  }
```


```
[info]   Cause: org.codehaus.janino.JaninoRuntimeException: failed to compile: org.codehaus.janino.JaninoRuntimeExc
eption: Constant pool for class org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificMutableProjection 
has grown past JVM limit of 0xFFFF
[info]   at org.apache.spark.sql.catalyst.expressions.codegen.CodeGenerator$.org$apache$spark$sql$catalyst$expressi
ons$codegen$CodeGenerator$$doCompile(CodeGenerator.scala:1079)
```

Reply:
thank you very much for your help @viirya ! In my use cases it seemed to be connected to the `dropDuplicates` method and I focused on it, but thanks to your suggestion now I realize that `dropDuplicates` by itself is not enough, it needs also some functions applied to columns to generate the issue! Thank you so much. Where should I add this test case?

Reply:
I was adding it to `DataFrameAggregateSuite`.

Reply:
@viirya I have a good and a bad news... Thanks to your suggestion I have been able to understand and reproduce the issue. Moreover, I found also another issue which is fixed by this problem and I am adding a UT for that too: in some cases, we might have a 
```
Code of method apply(...) grows beyond 64 KB
```
And with this PR the problem is fixed.

The bad thing is that the UT you provided still fails, but with a different error: actually it is always a Constant Pool limit exceeded exception, but it is in a NestedClass. From my analysis, this is caused by another problem, ie. that we might reference too many fields of the superclass in the NestedClasses. This might be addressed maybe trying to tune the magic number which I brought to 1000k in this PR, but I am pretty sure that it will be also addressed by the ongoing PR for SPARK-18016, since he is trying to reduce the number of variables. Thus I consider this out of scope for this PR.

Reply:
@mgaido91 Do you meant "2: compact primitive declarations into arrays" in SPARK-18016?

Reply:
@mgaido91 Thanks for trying it. Yeah, those expressions like `skewness` are very complicated, so they're likely to cause the issue you encountered.

Reply:
@viirya exactly, I meant that. Thank you for your suggestion. You have been very helpful to me.